cx_Freeze/freezer.py

```diff
@@ -267,45 +191,13 @@ def _FreezeExecutable(self, exe):
 
         # Search the C runtimes, using the directory of the python libraries
         # and the directories of the base executable
-        if sys.platform == "win32" and self.runtime_files:
-            search_dirs = set()
-            for filename in dependent_files:
-                search_dirs.add(os.path.split(filename)[0])
-            for filename in self.runtime_files:
-                for search_dir in search_dirs:
-                    filepath = os.path.join(search_dir, filename)
-                    if os.path.exists(filepath):
-                        dependent_files.add(filepath)
-
-        # Always copy the dynamic libraries into lib folder
-        target_dir = os.path.join(self.targetdir, "lib")
+        self._PlatformAddExtraExecutableDependencies(dependent_files=dependent_files)
+
         for source in dependent_files:
-            # but python dlls should be copied within the executable
-            if sys.platform == "win32" and source.endswith(python_libs):
-                target = os.path.join(self.targetdir, os.path.basename(source))
-            else:
-                target = os.path.join(target_dir, os.path.basename(source))
-            if sys.platform == "darwin":
-                # this recovers the cached MachOReference pointers to the files
-                # found by the _GetDependentFiles calls above. If one is found,
-                # pass into _CopyFile.
-                # We need to do this so the file knows what file referenced it,
-                # and can therefore calculate the appropriate rpath.
-                # (We only cache one reference.)
-                cachedReference = self.darwinTracker.getCachedReferenceTo(
-                    sourcePath=source
-                )
-                self._CopyFile(
-                    source,
-                    target,
-                    copyDependentFiles=True,
-                    includeMode=True,
-                    machOReference=cachedReference,
-                )
-            else:
-                self._CopyFile(
-                    source, target, copyDependentFiles=True, includeMode=True
-                )
+            # Store dynamic libraries in appropriate location for platform
+            target = os.path.join(self._PlatformExecutableDependencyDir(), os.path.basename(source))
```

Comment:
I am in doubt if the method _PlatformExecutableDependencyDir needs to exist. You can restore the line 
target_dir = os.path.join(self.targetdir, "lib")
before the for and take care of the correct path in _CopyTopDependency in WinFreezer

Reply:
Okay, that makes sense.  I eliminated _PlatformExecutableDependencyDir and moved the code into a new _CopyTopDependency for WinFreezer instead.